src/hotspot/share/gc/z/zNUMA.cpp

```diff
@@ -24,18 +24,28 @@
 #include "precompiled.hpp"
 #include "gc/shared/gcLogPrecious.hpp"
 #include "gc/z/zNUMA.hpp"
+#include "gc/z/zNUMA.inline.hpp"
 
-bool ZNUMA::_enabled;
+ZNUMA::State ZNUMA::_state;
 
 void ZNUMA::initialize() {
   pd_initialize();
 
   log_info_p(gc, init)("NUMA Support: %s", to_string());
-  if (_enabled) {
+  if (is_enabled()) {
     log_info_p(gc, init)("NUMA Nodes: %u", count());
   }
 }
 
 const char* ZNUMA::to_string() {
-  return _enabled ? "Enabled" : "Disabled";
+  switch (_state) {
```

Comment:
Maybe consider to have a `static const char* strings[] = {"Enabled", "Unsupported", "Disabled" };` (in the correct order) constant here and index by the array (and maybe an assert to catch out of bounds access if/when adds an entry and forgets to update this list).
This is typically more readable then a long switch case like this, but idk other people's opinions.

Reply:
And of course it's not a particular important suggestion, feel free to ignore.

Reply:
> Maybe consider to have a `static const char* strings[] = {"Enabled", "Unsupported", "Disabled" };` (in the correct order) constant here and index by the array (and maybe an assert to catch out of bounds access if/when adds an entry and forgets to update this list).
> This is typically more readable then a long switch case like this, but idk other people's opinions.

Thanks @tschatzl for your review.

I didn't think of this skill before.
Thanks for your teaching.

Let's just keep it this time since there are only 3 cases for `_state`.
And I'd like to use your style of code next time.

Thanks.
Best regards,
Jie

Reply:
I'd like to keep the current proposal, since it's more inline with what we do in other places in ZGC, like `ZLargePages::to_string()`.